test/cmd/router.sh

```diff
@@ -37,6 +37,10 @@ os::cmd::expect_success_and_not_text "oadm router --dry-run --host-network=false
 os::cmd::expect_success_and_not_text 'oadm router --dry-run --host-network=false --host-ports=false -o yaml' 'OPENSHIFT_MASTER'
 os::cmd::expect_success_and_not_text 'oadm router --dry-run --host-network=false --host-ports=false --secrets-as-env -o yaml' 'OPENSHIFT_MASTER'
 os::cmd::expect_success_and_text 'oadm router --dry-run --host-network=false --host-ports=false --secrets-as-env --credentials=${KUBECONFIG} -o yaml' 'OPENSHIFT_MASTER'
+# canonical hostname
+os::cmd::expect_success_and_text 'oadm router --dry-run --router-canonical-hostname=a.b.c.d -o yaml' 'a.b.c.d'
+os::cmd::expect_failure_and_not_text 'oadm router --dry-run --router-canonical-hostname=1a.b.c.d -o yaml' '1a.b.c.d'
```

Comment:
@knobunc IsHostname() tests it that way. It also limits to 255 characters instead of 253. They seem to follow RFC 952. Do we want to add support for RFC 1123 in the vendor code? BTW: when the name can start with a digit, 1.1.1.1 becomes a valid name. You explicitly wanted to not allow IP addresses.

Are there any other places in openshift where hostnames are validated? If so, I can use that code. If not, why start here?